### DISCUSSION TITLE
Can't find cpeditor.desktop after building from source

### DISCUSSION THREAD

**Discussion by hiensumi** [cpeditor/cpeditor#1259] [Q&A] [answered]
![image](https://github.com/cpeditor/cpeditor/assets/47080203/83c8caf3-2e3c-4d18-91f7-7662cd37c97a)

**Comment by ouuan** [comment]
The desktop file is at https://github.com/cpeditor/cpeditor/blob/master/dist/linux/cpeditor.desktop

You can use `cmake --install` to install it.

**Reply by hiensumi** [reply]
I tried multiple times but cmake always install cpeditor at /usr/local/bin and the cpeditor.desktop is pointed to /usr/bin.
I attempted to edit the cpeditor.desktop exec path to /usr/local/bin but it is still not working.

**Comment by ouuan** [comment]
> I attempted to edit the cpeditor.desktop exec path to /usr/local/bin but it is still not working.

What is "not working"? BTW, which version are you compiling?

**Reply by hiensumi** [reply]
6.10.3 . When i click on cpeditor.desktop(after cmake --install .), it does not load the application.

**Reply by ouuan** [reply]
Can you start it via command line?

**Reply by hiensumi** [reply]
how?

**Reply by ouuan** [reply]
Just run the built `cpeditor`, i.e. `./cpeditor` or `build/cpeditor`. If you have installed it, then just `cpeditor`.

**Reply by hiensumi** [reply]
cpeditor: /lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.15' not found (required by cpeditor)
cpeditor: /lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.15' not found (required by /usr/local/lib/x86_64-linux-gnu/libKF5SyntaxHighlighting.so.5)
Uh oh

**Reply by hiensumi** [reply]
![image](https://github.com/cpeditor/cpeditor/assets/47080203/c464fe68-f586-4a5f-a91a-b777ace98b03)
Already have qt5.15.2 :v

**Reply by ouuan** [reply]
Is that Qt path in your `PATH` environment variable?

**Reply by hiensumi** [reply]
PATH=/home/hiensumi/Qt/5.15.2/gcc_64/bin:/home/hiensumi/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

Yes.

**Reply by ouuan** [reply]
`/bin` is in your PATH but `/lib` is not.

**Reply by hiensumi** [reply]
![image](https://github.com/cpeditor/cpeditor/assets/47080203/db929059-0cc9-432c-92c8-4afd3b71a4a3)
Gosh

**Reply by hiensumi** [reply]
@ouuan Maybe I installed qt5 the wrong way. How did you install it?

**Reply by ouuan** [reply, marked as answer]
I use Arch Linux.

It seems that your system Qt is picked in favor of the Qt in your home 🤔 Try setting `LD_LIBRARY_PATH`.

**Reply by hiensumi** [reply]
It works tysm :heart:

**Reply by hiensumi** [reply]
@ouuan How to make the .desktop file?
I have to use the terminal every time I want to launch cpeditor.

**Reply by ouuan** [reply]
`Exec=env LD_LIBRARY_PATH=/path /usr/local/bin/cpeditor`

**Reply by hiensumi** [reply]
@ouuan 
I got u but it launches and closes immediately

```
[Desktop Entry]
Version=1.0
Type=Application
Name=CP Editor
GenericName=Code editor for competitive programming
GenericName[ru]=Редактор кода для спортивного программирования
GenericName[zh_CN]=算法竞赛代码编辑器
GenericName[zh_TW]=競程程式碼編輯器
Exec=env LD_LIBRARY_PATH=/home/hiensumi/Qt/5.15.2/gcc_64/lib/ /usr/local/bin/cpeditor
Comment=CP Editor is a Qt-based, lightweight and cross-platform code editor specially designed for competitive programming.
Comment[RU]=CP Editor - это легкий кросс-платформенный редактор кода на базе Qt, специально разработанный для спортивного программирования.
Comment[zh_CN]=CP Editor 是一个基于 Qt 的轻量级跨平台代码编辑器，专为算法竞赛设计。
Comment[zh_TW]=CP Editor 是一款採用 Qt 的輕巧型跨平台程式碼編輯器，專為競程使用設計。
Icon=cpeditor
Terminal=false
StartupNotify=true
Encoding=UTF-8
Categories=Development;TextEditor;Qt;
MimeType=text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-python;
Path=/home/hiensumi


```

**Comment by ouuan** [comment]
> I got u but it launches and closes immediately

Did you see it actually launched?

**Reply by hiensumi** [reply]
Yes. It works fine with Terminal=true but not with false

**Reply by ouuan** [reply]
Are you sure you are building 6.10.3? Which commit is it?

**Comment by hiensumi** [comment]
> Are you sure you are building 6.10.3? Which commit is it?

I think im fine with the terminal running. Extra slot on the taskbar is ok

**Comment by ouuan** [comment]
Q&A discussions are resolved via marking answers, not closing.
